src/main/java/io/quarkus/gizmo/ClassSignatureBuilderImpl.java

```diff
@@ -0,0 +1,90 @@
+package io.quarkus.gizmo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.quarkus.gizmo.SignatureBuilder.ClassSignatureBuilder;
+import io.quarkus.gizmo.Type.ClassType;
+import io.quarkus.gizmo.Type.ParameterizedType;
+import io.quarkus.gizmo.Type.TypeVariable;
+
+class ClassSignatureBuilderImpl implements ClassSignatureBuilder {
+    private List<TypeVariable> typeParameters = new ArrayList<>();
+    private Type superClass = ClassType.OBJECT;
+    private List<Type> superInterfaces = new ArrayList<>();
+
+    @Override
+    public String build() {
+        StringBuilder signature = new StringBuilder();
+
+        // type params
+        if (!typeParameters.isEmpty()) {
+            signature.append('<');
+            for (TypeVariable typeParameter : typeParameters) {
+                typeParameter.appendTypeParameterToSignature(signature);
+            }
+            signature.append('>');
+        }
+
+        // superclass
+        superClass.appendToSignature(signature);
+
+        // interfaces
+        if (!superInterfaces.isEmpty()) {
+            for (Type superInterface : superInterfaces) {
+                superInterface.appendToSignature(signature);
+            }
+        }
+        return signature.toString();
+    }
+
+    @Override
+    public ClassSignatureBuilder addTypeParameter(TypeVariable typeParameter) {
+        typeParameters.add(typeParameter);
+        return this;
+    }
+
+    @Override
+    public ClassSignatureBuilder setSuperClass(ClassType superClass) {
+        this.superClass = superClass;
+        return this;
+    }
+
+    @Override
+    public ClassSignatureBuilder setSuperClass(ParameterizedType superClass) {
+        if (containsWildcard(superClass)) {
+            throw new IllegalArgumentException("A super type may not specify a wilcard");
+        }
+
+        this.superClass = superClass;
+        return this;
+    }
+
+    @Override
+    public ClassSignatureBuilder addSuperInterface(ClassType interfaceType) {
+        superInterfaces.add(interfaceType);
+        return this;
+    }
+
+    @Override
+    public ClassSignatureBuilder addSuperInterface(ParameterizedType interfaceType) {
+        if (containsWildcard(interfaceType)) {
+            throw new IllegalArgumentException("A super type may not specify a wilcard");
```

Comment:
Maybe this could use some tuning if we adjust `SuperInterface`

Reply:
You mean the message? Hm, so the `SuperInterface` is also a "super type" and the stack trace should contain enough information IMO. But again, if we decide to `A super interface may not specify a wilcard` then OK ;-)

Reply:
We could change this error message (and the other one in the `setSuperClass`) a bit I guess, even if it's quite precise as is. Something like "An implemented interface type may not specify a wildcard" or so (similarly, "an extended class type may not specify a wildcard").

Reply:
This wasn't addressed IIUC? Not a big deal in my opinion, just noting.

Reply:
Just an oversight... will fix it shortly.